src/CEOFESABundle/Controller/StructureController.php

```diff
@@ -239,4 +241,28 @@ private function createDeleteForm($id)
             ->getForm()
         ;
     }
+
+    /**
+     * @param $id
+     *
+     * @Route("/editpdf/{id}", name="structure_pdf")
+     * @return Response
+     */
+    public function createPdf($id)
+    {
+        $em = $this->getDoctrine()->getManager();
+
+        $entity = $em->getRepository('CEOFESABundle:Structure')->find($id);
+
+        $html = $this->renderView('::Templates\convention_partenariat.html.twig', array(
+            'info' => $entity
+        ));
+
+        $response= new \Symfony\Component\HttpFoundation\Response();
+        $response->setContent($this->get('knp_snappy.pdf')->getOutputFromHtml($html,array('orientation' => 'Portrait','page-size' => 'A4')));
+        $response->headers->set('Content-Type', 'application/pdf');
+        $response->headers->set('Content-disposition', 'filename=convention-partenariat-'.str_replace(" ", "_",$entity->getStrNom()).'.pdf');
```

Comment:
CS
& I'd prefer if you properly sluggify the name. We use Gedmo elsewhere so either use the Sluggable on Structure, Either use Gedmo's Urlizer :)